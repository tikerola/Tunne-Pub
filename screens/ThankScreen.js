import React from 'react';
import { Dimensions, StyleSheet, Text, View } from 'react-native';
import { colors } from '../constants/colors';
import { dimensions } from '../constants/dimensions';


export default function ThankScreen() {

  const { width } = Dimensions.get('window')


  return (
    <View style={{ flex: 1, alignItems: 'center', backgroundColor: colors.bodyBg }}>
      <View style={styles.container}>
        <View style={[styles.textContainer, { width: width <= dimensions.phoneMaxWidth ? '100%' : '75%' }]}>
          <Text style={styles.title}>Onneksi olkoon!</Text>
          <Text style={styles.text}>Olet nyt tallentanut vastauksesi ja voit alavalikon
          Tallennetut-napista valita tallennetun istunnon, tutkailla sitä ja tarpeen tullen
          tehdä lisämuutoksia.
        </Text>
        </View>
      </View>
    </View>

  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    padding: 10,
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center'
  },

  textContainer: {
    backgroundColor: 'rgba(0, 0,0, 0.8)',
    borderRadius: 8,
    padding: 15
  },

  backgroundImage: {
    flex: 1,
    resizeMode: "cover",
    justifyContent: "center"
  },

  title: {
    fontSize: 20,
    fontFamily: 'Roboto-Light',
    marginBottom: 25,
    color: '#bbb',
    textAlign: 'center'
  },

  text: {
    color: '#bbb',
    fontSize: 18,
    fontFamily: 'Roboto-Light',
    marginBottom: 30
  }

})
