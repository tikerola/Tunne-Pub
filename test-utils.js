import React from "react";
import { NavigationContext } from "@react-navigation/native";
import { render } from "@testing-library/react-native";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import CARD_DATA from "./data/cards";
import NEEDS_DATA from "./data/needs";
import { SENTENCES_DATA } from "./data/sentences";
import WORD_DATA from "./data/words";

const INITIAL_STATE = {
  cards: {
    // cards: CARD_DATA,
    // words: WORD_DATA,
    // needs: NEEDS_DATA,
    // sentences: SENTENCES_DATA,
    activeSummaryItemId: "123",
    summaryOfChosenItems: {
      123: {
        id: "123",
        name: "",
        cards: [],
        feelingsWords: [],
        feelingsYouMissWords: [],
        needs: ["hei"],
        needsBehindAFeeling: [],
        sentences: [],
        questions: {},
        itemSaved: true,
      },
    },
    screenNumber: 1,
    numOfItems: 0,
    initialLoadFinished: false,
  },
};

const mockStore = configureStore([]);
const store = mockStore(INITIAL_STATE);

const navContext = {
  isFocused: () => true,
  // addListener returns an unscubscribe function.
  addListener: jest.fn(() => jest.fn()),
};

const reduxRender = (ui, options, storeData = store) =>
  render(ui, {
    wrapper: ({ children }) => (
      <Provider store={storeData}>
        <NavigationContext.Provider value={navContext}>
          {children}
        </NavigationContext.Provider>
      </Provider>
    ),
    ...options,
  });

// re-export everything for convenience
export * from "@testing-library/react-native";

// override render method
export { reduxRender as render, store };
