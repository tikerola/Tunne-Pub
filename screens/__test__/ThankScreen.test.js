import React from "react";
import { render } from "../../test-utils";
import ThankScreen from "../ThankScreen";

describe("<ThankScreen />", () => {
  it("should match snapshot", () => {
    expect(render(<ThankScreen />)).toMatchSnapshot();
  });
});
