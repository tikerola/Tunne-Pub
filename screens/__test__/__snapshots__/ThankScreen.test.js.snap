// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThankScreen /> should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#5f0c38",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "padding": 10,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgba(0, 0,0, 0.8)",
            "borderRadius": 8,
            "padding": 15,
          },
          Object {
            "width": "75%",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#bbb",
            "fontFamily": "Roboto-Light",
            "fontSize": 20,
            "marginBottom": 25,
            "textAlign": "center",
          }
        }
      >
        Onneksi olkoon!
      </Text>
      <Text
        style={
          Object {
            "color": "#bbb",
            "fontFamily": "Roboto-Light",
            "fontSize": 18,
            "marginBottom": 30,
          }
        }
      >
        Olet nyt tallentanut vastauksesi ja voit alavalikon Tallennetut-napista valita tallennetun istunnon, tutkailla sitä ja tarpeen tullen tehdä lisämuutoksia.
      </Text>
    </View>
  </View>
</View>
`;
